internal/bits/bit_array_test.go

```diff
@@ -301,3 +387,15 @@ func TestUnmarshalJSONDoesntCrashOnZeroBits(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, ic.BitArray, &BitArray{Bits: 0, Elems: nil})
 }
+
+func BenchmarkPickRandomBitArray(b *testing.B) {
```

Comment:
I wonder if adding some contention to this benchmark would affect the results. I think the results are better with this PR, but maybe this is something to explore. By contention, I mean, another thread changing the bit set while we are retrieving random indexes.

Reply:
Where this is used, these mutexes are truly useless. Was going to make an issue to make a new bit array version that gets rid of the mutexes

Reply:
This would be ideal. In particular because I assume that the methods using this are already synchronized.

For instance, consensus state, peer consensus state should be synchronized.